packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1586,8 +1577,27 @@ class DialogRoute<T> extends RawDialogRoute<T> {
          },
          barrierLabel: barrierLabel ?? MaterialLocalizations.of(context).modalBarrierDismissLabel,
          transitionDuration: const Duration(milliseconds: 150),
-         transitionBuilder: _buildMaterialDialogTransitions,
        );
+
+  CurvedAnimation? _fadeCurve;
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    _fadeCurve ??= CurvedAnimation(
+      parent: animation,
+      curve: Curves.easeOut,
+    );
```

Comment:
Is it possible that `buildTransitions` is called a 2nd time with an `animation` parameter that is not the same as the one when it was first called? If yes, the `CurvedAnimation _fadeCurve` needs to be replaced with a new one to have the correct animation

Reply:
Thanks for pointing it out, i have looked into `abstract class ModalRoute` and it initialises animation only once and it does not get replaced so using above code snippet is safe :)